x/profiles/keeper/keeper_chain_link.go

```diff
@@ -0,0 +1,148 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	sdkquery "github.com/cosmos/cosmos-sdk/types/query"
+
+	"github.com/desmos-labs/desmos/x/profiles/types"
+)
+
+// StoreChainLink stores the given chain link inside the current context.
+// It assumes that the given chain link has already been validated.
+func (k Keeper) StoreChainLink(ctx sdk.Context, link types.ChainLink, destinationAddress string) error {
+	if _, found := k.GetChainLink(ctx, link.ChainConfig.Name, link.Address); found {
+		return fmt.Errorf("chain link already exists")
+	}
+
+	// Check if address has the profile
```

Comment:
I think there might be a missing check here. We should also check if the recipient address has a profile as well. This can be done by reading the address value using either the `sdk.AccAddressFromBech` or the `sdk.AccAddressFromHex` methods, and seeing if the associated Desmos account has a profile. 

As we discussed previously, we must avoid connecting together two Desmos profiles.